proxy/frontend/core/src/main/java/org/apache/shardingsphere/proxy/frontend/netty/CDCChannelInboundHandler.java

```diff
@@ -164,22 +201,27 @@ private void processStartSubscription(final ChannelHandlerContext ctx, final CDC
                     .addListener(ChannelFutureListener.CLOSE);
             return;
         }
+        checkPrivileges(connectionContext.getCurrentUser().getGrantee(), startSubscriptionRequest.getDatabase());
         CDCResponse response = backendHandler.startSubscription(request, ctx.channel(), connectionContext);
         ctx.writeAndFlush(response);
     }
     
-    private void stopStartSubscription(final ChannelHandlerContext ctx, final CDCRequest request, final CDCConnectionContext connectionContext) {
+    private void processStopSubscription(final ChannelHandlerContext ctx, final CDCRequest request, final CDCConnectionContext connectionContext) {
+        StopSubscriptionRequest stopSubscriptionRequest = request.getStopSubscription();
+        checkPrivileges(connectionContext.getCurrentUser().getGrantee(), stopSubscriptionRequest.getDatabase());
         backendHandler.stopSubscription(connectionContext.getJobId());
         connectionContext.setStatus(CDCConnectionStatus.LOGGED_IN);
         ctx.writeAndFlush(CDCResponseGenerator.succeedBuilder(request.getRequestId()).build());
     }
     
-    private void dropStartSubscription(final ChannelHandlerContext ctx, final CDCRequest request, final CDCConnectionContext connectionContext) {
+    private void processDropSubscription(final ChannelHandlerContext ctx, final CDCRequest request, final CDCConnectionContext connectionContext) {
+        DropSubscriptionRequest dropSubscriptionRequest = request.getDropSubscription();
+        checkPrivileges(connectionContext.getCurrentUser().getGrantee(), dropSubscriptionRequest.getDatabase());
         try {
             backendHandler.dropSubscription(connectionContext.getJobId());
             ctx.writeAndFlush(CDCResponseGenerator.succeedBuilder(request.getRequestId()).build());
         } catch (final SQLException ex) {
-            ctx.writeAndFlush(CDCResponseGenerator.failed(request.getRequestId(), CDCResponseErrorCode.SERVER_ERROR, ex.getMessage()));
+            ctx.writeAndFlush(CDCResponseGenerator.failed(request.getRequestId(), CDCResponseErrorCode.SERVER_ERROR, ex.getMessage())).addListener(ChannelFutureListener.CLOSE);
         }
```

Comment:
Is it necessary to close connection when internal exception occur when drop subscription

Reply:
Now it is not clear whether to close the channel when the exception occurs.
Perhaps we can assume that any exceptions before successful login will close the channel, and any exceptions after successful login will not close the channel

Reply:
OK. We could add TODO in code for now